ChelseasFieldJournal/src/GUI/GuiComposerHelpers.cs

```diff
@@ -0,0 +1,836 @@
+        }
+
+        public static GuiElementIconListPicker GetIconListPicker(this GuiComposer composer, string key)
+        {
+            return (GuiElementIconListPicker)composer.GetElement(key);
+        }
+
+        public static void IconListPickerSetValue(this GuiComposer composer, string key, int selectedIndex)
+        {
+            int i = 0;
+            GuiElementIconListPicker btn;
+            while ((btn = composer.GetIconListPicker(key + "-" + i.ToString())) != null)
+            {
+                btn.SetValue(i == selectedIndex);
+                i++;
+            }
+        }
+
+        public static GuiComposer AddIconListPicker(this GuiComposer composer, string[] icons, Action<int> onToggle, ElementBounds startBounds, int maxLineWidth, string key = null)
+        {
+            return composer.AddElementListPicker(typeof(GuiElementIconListPicker), icons, onToggle, startBounds, maxLineWidth, key);
+        }
+
+        public static GuiComposer AddVerticalScrollbar(this GuiComposer composer, Action<float> onNewScrollbarValue, ElementBounds bounds, string key = null)
+        {
+            if (!composer.Composed)
+            {
+                composer.AddInteractiveElement(new GuiElementScrollbar(composer.Api, onNewScrollbarValue, bounds), key);
+            }
+            return composer;
+        }
+
+        public static GuiElementScrollbar GetScrollbar(this GuiComposer composer, string key)
+        {
+            return (GuiElementScrollbar)composer.GetElement(key);
+        }
+
+        public static GuiComposer AddSlider(this GuiComposer composer, ActionConsumable<int> onNewSliderValue, ElementBounds bounds, string key = null)
+        {
+            if (!composer.Composed)
+            {
+                composer.AddInteractiveElement(new GuiElementSlider(composer.Api, onNewSliderValue, bounds), key);
+            }
+            return composer;
+        }
+
+        public static GuiElementSlider GetSlider(this GuiComposer composer, string key)
+        {
+            return (GuiElementSlider)composer.GetElement(key);
+        }
+
+        public static GuiComposer AddSwitch(this GuiComposer composer, Action<bool> onToggle, ElementBounds bounds, string key = null, double size = 30.0, double padding = 4.0)
+        {
+            if (!composer.Composed)
+            {
+                composer.AddInteractiveElement(new GuiElementSwitch(composer.Api, onToggle, bounds, size, padding), key);
+            }
+            return composer;
+        }
+
+        public static GuiElementSwitch GetSwitch(this GuiComposer composer, string key)
+        {
+            return (GuiElementSwitch)composer.GetElement(key);
+        }
+
+        [Obsolete("Use Method without orientation argument")]
+        public static GuiComposer AddButton(this GuiComposer composer, string text, ActionConsumable onClick, ElementBounds bounds, CairoFont buttonFont, EnumButtonStyle style, EnumTextOrientation orientation, string key = null)
```

Comment:
If it's obsolete and we provide an alternative, remove the obsolete methods entirely.